src/aap_eda/services/ruleset/activate_rulesets.py

```diff
@@ -113,10 +113,30 @@ def activate(self, activation: models.Activation) -> None:
         ssl_verify = settings.WEBSOCKET_SSL_VERIFY
         try:
             try:
+                project_hash = None
```

Comment:
@Dostonbek1 With the current implementation when every rulebook has a project couldn't we use this all the time
```
activation.rulebook.project.git_hash
```
This would have the latest hash and it would get saved into the new ActivationInstance we create.

This works even when you have don't have the project is None since we are getting the data from the rulebook project relationship.

Reply:
We have to have this check here for when we delete the project after creating an activation. Although the project is deleted, rulebook content is still saved into activation and activation can be restarted. In that case, rulebook will be null.

Reply:
@Dostonbek1 I think it might make sense to store the git_hash at both the Activation and ActivationInstance level. The Activation git_hash can be updated with every project_sync by updating it [here](https://github.com/ansible/eda-server/blob/249bd8e586a69bf4b601b4917db8ed329f867ad2/src/aap_eda/services/project/imports.py#L171)
The ActivationInstance git_hash will be copied from the Activation git_hash when we run a new activation. This way even if the project gets deleted we would have the last git_hash in the Activation object. The 2 git hashes have to be displayed in 2 different screens.
The Activation git_hash should be visible here
<img width="965" alt="Screenshot 2023-08-29 at 3 43 39 PM" src="https://github.com/ansible/eda-server/assets/6452699/d33c5450-299d-4264-b061-3dde4a3cb5ca">

And the Activation Instance git_hash should be available for every activation run here
<img width="1232" alt="Screenshot 2023-08-29 at 3 44 10 PM" src="https://github.com/ansible/eda-server/assets/6452699/17fc0dad-b5ca-4056-83e7-118d0cef2474">

Reply:
@mkanoor I think the activation hash should be the hash from the latest instance if exists. We should take into account the case where we have an instance running with hash B, but project was updated later and the activation is showing the  hash C, which would be incorrect. So sync_project task should update the hash only on the activations that don't have any instance running. What do you think?

Reply:
@Alex-Izquierdo I am not sure I follow, the Activation.git_hash = <latest project hash>
ActivationInstance.git_hash = Activation.git_hash
These are not references they are the real values, the Activation gets the value from project_sync and the ActivationInstances gets from Activation just before a new Activation starts. They can deviate after this but each one has the git_hash value of their own. We are always copying the value from the Activation into the ActivationInstance. 

In this case both the ActivationInstance and Activation have an attribute called git_hash. The ActivationInstance.git_hash will never be updated after it has been created. The Activation.git_hash might keep changing with each project sync if there are differences.

Reply:
I mean that activation.git_hash should have the value of the latest activation_instance.git_hash if the instances exists. 
If we update the activation.git_hash from sync_project, it might happen that the activation.git_hash is different from the latest activation_instance.git_hash. That case must be taken into account.

Reply:
The activation.git_hash will have the latest information from the project, so even if the project is deleted after that sync we have captured the latest hash. The ActivationInstance.git_hash tells you which hash was used for that activation run. If the Activation.git_hash != ActivationInstance.git_ash (latest run) then we know that we are running with an old rulebook. If we set Activation.git_hash = ActivationInstance.git_ash (latest run) then we don't have the latest project hash. The 2 screens above convey different things we would have to add a column to the History to display the git_hash for the Activation Instance. The git_hash in the Activation Instance is what the latest value is. If you want to capture what was the last run git_hash we would have to then add
* Activation.git_hash
* Activation.last_run_git_hash
* ActivationInstance.git_hash

I think the last_run_git_hash is redundant since you can get that from the latest instance and the UI can display that in the History as an additional column and be color coded if its different from the Activation.git_hash so you could spot that there is something odd with that ActivationInstance.

Reply:
Another thing should this git_hashes be clickable/link and they would show you the tree in the git repo which has the corresponding hash.

Reply:
Your point about `Activation.git_hash != ActivationInstance.git_hash` is valid but I think the user expectation is to see for an activation that is running the hash of the latest instance.

Reply:
> Another thing should this git_hashes be clickable/link and they would show you the tree in the git repo which has the corresponding hash.

@mkanoor Is this a question or affirmation? The UI mock currently is defined only to provide the string. What you propose would be a new feature. We should need to know in advance what would be the url, it depends on the git server (gitea, github, gitlab...). AWX also provides only the string.

Reply:
FYI, JIRA ticket for UI changes that Madhu mentioned: [AAP-15544](https://issues.redhat.com/browse/AAP-15544)